src/ecdsapod.rs

```diff
@@ -210,10 +213,45 @@ impl middleware::RecursivePod for EcdsaPod {
         self.proof.clone()
     }
     fn vds_root(&self) -> Hash {
-        self.vds_hash
+        self.vds_root
     }
 }
 
+impl Pod for EcdsaPod {
+    fn params(&self) -> &Params {
+        &self.params
+    }
+    fn verify(&self) -> Result<(), Box<DynError>> {
+        Ok(self._verify().map_err(Box::new)?)
+    }
+
+    fn id(&self) -> PodId {
+        self.id
+    }
+
+    fn pod_type(&self) -> (usize, &'static str) {
+        (PodType::Ecdsa as usize, "Ecdsa")
+    }
+
+    fn pub_self_statements(&self) -> Vec<middleware::Statement> {
+        pub_self_statements(self.msg, self.pk)
+    }
+
+    fn serialize_data(&self) -> serde_json::Value {
```

Comment:
The goal of this method is to serialize the pod, so that later on we can deserialize it!

Could you also add deserialization methods to the eddsa and ed25519 pod?
Here's an example of deserialization in the MainPod (you should use the same function signature): https://github.com/0xPARC/pod2/blob/3c6930dfe61ce28f925bea942dcff839c910ddc7/src/backends/plonky2/mainpod/mod.rs#L647-L664

You may decide to change the data that you're serializing.  I think it will be easier if you include the `msg` and `pk` instead of the public statements.

It would also be great if you add a test where you serialize and deserialize and check that the pod is equal before and after.

Reply:
Makes sense, I wonder then, if we should add the `deserialize` method into the `Pod` trait, to enforce that it exists in future implementations of pods (to not depend on reviews detecting it missing it).

Reply:
That's a good point.  Originally I didn't add this method to the `Pod/RecursivePod` traits because these traits are used for trait objects.  And in a trait object I don't think you can access a method that doesn't use `self`, and the deserialize method doesn't use `self`.

But as you point out, having this method in the trait will enforce the developer of an introduction pod to implement it!

Reply:
I've created an issue for this https://github.com/0xPARC/pod2/issues/294

Reply:
I've added the deserializer (https://github.com/0xPARC/introduction-pods/pull/4/commits/594cb58ad286817811fdd3528b17cd4af64eae02), but I think I'm going put back to draft the current PR and then implement https://github.com/0xPARC/pod2/issues/294 and update the current PR to use the new (future) pod2 version (also adding a test).

Reply:
Update: https://github.com/0xPARC/pod2/pull/300 (which solves https://github.com/0xPARC/pod2/issues/294 ) has been merged, I've updated the current PR to latest pod2 version including those changes.